src/mimeparser.rs

```diff
@@ -1303,13 +1304,26 @@ fn get_attachment_filename(mail: &mailparse::ParsedMail) -> Result<Option<String
                 }
             });
         if apostrophe_encoded {
-            // we're currently always assuming utf-8, this might need adaption, however, should not break things.
             if let Some(name) = desired_filename {
-                desired_filename = if let Some(name) = name.splitn(3, '\'').last() {
-                    Some(percent_decode_str(&name).decode_utf8_lossy().to_string())
-                } else {
-                    None
-                }
+                let mut parts = name.splitn(3, '\'');
+                desired_filename =
+                    if let (Some(charset), Some(value)) = (parts.next(), parts.last()) {
+                        let decoded_bytes = percent_decode_str(&value);
+                        if charset.to_lowercase() == "utf-8" {
+                            Some(decoded_bytes.decode_utf8_lossy().to_string())
+                        } else {
+                            let charset = match Charset::for_label(charset.as_bytes()) {
```

Comment:
I would rather use something like
```
if let Some(charset) = Charset::for_label(charset.as_bytes()).or_else(|| Charset::for_label(b"latin1")) {
  ...
} else {
  warn!(context, "latin1 encoding does not exist");
  None
}
```
to avoid `unwrap()` even though we test for it. We don't run tests on target platforms, so if `latin1` does not exist there for some reason after cross-compilation, we will not get a panic.

Reply:
great, i was also a bit unhappy about the unwrap (therefore the test), but i did not found a "nice" solution.

i also added some `warn!`, makes sense, however, i needed to add `Context` to get_attachment_filename() for that purpose, therefore the larger change.